lib/discourse_tagging.rb

```diff
@@ -296,12 +313,23 @@ def self.filter_allowed_tags(guardian, opts = {})
     end
 
     if filter_for_non_staff
-      # remove hidden tags
-      builder.where(<<~SQL, Group::AUTO_GROUPS[:everyone])
+      group_ids = permitted_group_ids(guardian)
+
+      builder.where(<<~SQL, group_ids, group_ids)
         id NOT IN (
-          SELECT tag_id
-            FROM tag_group_memberships tgm
-           WHERE tag_group_id NOT IN (SELECT tag_group_id FROM tag_group_permissions WHERE group_id = ?)
+          (SELECT tgm.tag_id
+             FROM tag_group_permissions tgp
+       INNER JOIN tag_groups tg ON tgp.tag_group_id = tg.id
```

Comment:
The indentation here looks a bit off.

Reply:
The way the various chunks of hand-written SQL is formatted in this file is extremely inconsistent. Do we have a set of rules/style guide on how it should be done?

Reply:
I agree it's inconsistent, but maybe this is displaying differently for you. Check this out:

![image](https://user-images.githubusercontent.com/17538/95489660-2a3e3980-0965-11eb-8815-b273b53582b0.png)

The `INNER JOIN`s are to the left of the `(SELECT`, that they belong to. I feel like any parts of a subquery should be to the right of where it starts, or at least aligned in a column. The `FROM` and `WHERE` seem good.